src/components/include/DefaultImportModalDialog.jsx

```diff
@@ -0,0 +1,152 @@
+            ...material.lattice,
+            ...{
+                a: mockLatticeConstant,
+                b: mockLatticeConstant,
+                c: mockLatticeConstant,
+            },
+        };
+        const lattice = new Made.Lattice(latticeConfig);
+
+        const basis = new Made.Basis({
+            ...basisConfig,
+            cell: lattice.vectorArrays,
+        });
+
+        const newMaterialConfig = {
+            ...material.toJSON(),
+            ...{
+                basis: basis.toJSON(),
+                lattice: lattice.toJSON(),
+                name: `${this.reader.currentFilename} - ${basis.formula}`,
+            },
+        };
+        const newMaterial = new Material(newMaterialConfig);
+        newMaterial.cleanOnCopy();
+
+        Made.tools.material.getBasisConfigTranslatedToCenter(newMaterial);
+        this.props.onSubmit([newMaterial]);
+    }
+
+    handleChange(content) {
+        this.setState({ xyz: content });
+    }
+
+    handleFileChange(files) {
+        if (!files[0] || !files[0].size)
+            return NPMsAlert.warning("Error: file cannot be read (unaccessible?)");
+        this.reader.currentFilename = files[0].name;
+        this.reader.readAsText(files[0]);
+    }
+
+    renderHeader() {
+        return (
+            <ModalHeader className="bgm-dark" closeButton>
+                <h4 className="modal-title">{this.props.title}</h4>
+            </ModalHeader>
+        );
+    }
+
+    renderBody() {
+        return (
+            <ModalBody className="bgm-dark">
+                <BasisText
+                    ref={(el) => {
+                        this.BasisTextComponent = el;
+                    }}
+                    className="default-import-modal-basis-text"
+                    content={this.state.xyz}
+                    onChange={this.handleChange}
+                />
+                <input
+                    type="file"
+                    id="fileapi"
+                    onChange={(event) => this.handleFileChange(event.target.files)}
+                />
+
+                <div className="row m-t-10">
+                    <div className="col-md-12">
+                        <button
+                            id="default-import-modal-submit"
+                            className="btn btn-custom btn-block"
+                            type="button"
+                            onClick={this.handleSubmit}
+                        >
+                            Submit your XYZ
+                        </button>
+                    </div>
+                </div>
+            </ModalBody>
+        );
+    }
+
+    // TODO: resolve the linter issue:
+    //   Line 150:17:  Expected 'this' to be used by class method 'renderFooter'  class-methods-use-this
+    // renderFooter() {
+    //     return null;
+    // }
+}
+
+DefaultImportModalDialog.PropTypes = {
+    onSubmit: PropTypes.func,
+    material: PropTypes.object,
+};
+
+export default DefaultImportModalDialog;
```

Comment:
We should consider a test for this functionality, similar to https://github.com/Exabyte-io/materials-designer/tree/dev/tests/cucumber/features/menu and including some of the fixtures you provided earlier in this PR